src/System.Buffers/src/System/Buffers/BufferPool.cs

```diff
@@ -41,7 +36,8 @@ public T[] RentBuffer(int size, bool clearBuffer = false)
             }
             else
             {
-                buffer = _buckets[SelectBucketIndex(size)].Rent();
+                int index = SelectBucketIndex(size);
+                buffer = LazyInitializer.EnsureInitialized(ref _buckets[index], () => new BufferBucket<T>(GetMaxSizeForBucket(index), NUMBER_OF_BUFFERS_IN_BUCKET)).Rent();
```

Comment:
I looked at this for awhile and I'm not entirely sure how I can avoid both the closure and delegate allocations here, at least without entirely refactoring the BufferBucket class. Since the BufferBucket.ctor call requires at least one unique value, the value to place into there needs to be captured by the delegate to be passed to the ctor when the delegate fires. If you guys know a way around this without drastic refactors, I'm open to ideas :)